kubernetes/terraform/modules/kubernetes/external_dns.tf

```diff
@@ -116,25 +103,29 @@ resource "kubernetes_deployment" "external_dns" {
           "app"    = "external-dns",
         }
         annotations = {
-          "iam.amazonaws.com/role" = "k8s-external-dns-role",
+          "eks.amazonaws.com/role-arn" = module.iam_assumable_role.this_iam_role_arn
         }
       }
       spec {
         container {
           name  = "external-dns"
           image = "registry.opensource.zalan.do/teapot/external-dns:latest"
           args = [
-            "--source=service",
             "--source=ingress",
             "--domain-filter=${var.external_dns_zone}", # Give access only to the specified zone
             "--provider=aws",
             "--aws-zone-type=public",
             "--policy=upsert-only", # Prevent ExternalDNS from deleting any records
             "--registry=txt",
-            "--txt-owner-id=${var.external_dns_owner_id}", # ID of txt record to manage state
+            "--txt-owner-id=${var.cluster_name}", # ID of txt record to manage state
+            "--aws-batch-change-size=2", # Set the batch size to 2 so that a single record failure won't block other updates
           ]
         }
 
+        security_context {
+          fs_group =  65534
```

Comment:
Required for applications inside the container to be able to access the token file that gets mounted